### PR TITLE
CI Build & Test: Fix test errors involving optional packages coxeter3, ...

### PR DESCRIPTION
**Pull request by mkoeppe**
These errors have started to show up after #35661 was merged.

Optional packages such as `coxeter3` are installed as a side effect of testing distributions such as sagemath-coxeter.
Hence the doctests marked `# optional - coxeter3` are activated.
Here we ensure that the corresponding extension modules are also installed in the venv where the doctests run.

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -103,12 +103,24 @@ jobs:
...
+          SAGE_NUM_THREADS: 2
+
+      - name: Build
+        # This step is needed because building the modularized distributions installs some optional packages,
+        # so the editable install of sagelib needs to build the corresponding optional extension modules.
+        id: build
+        if: always() && (steps.incremental.outcome == 'success' || steps.clean.outcome == 'success')
+        run: |
+          make build
         working-directory: ./worktree-image
```
Comment (kwankyu) [inline .github/workflows/build.yml:124]: You just split `make build` from the `Clean` step. How does this make it sure that the "corresponding extension modules" are installed? Prior to that, what do you mean by "corresponding extension modules"? Sorry for naive questions.

Comment (mkoeppe) [reply .github/workflows/build.yml:124]: The "Clean" step is only run when something went wrong in the incremental build.

By splitting out the "make build" step from, this step gets run unconditionally.

In an editable build (default), the Sage library is monolithic: "make sagelib" is responsible for installing all of the Sage library, including the extension modules (Cython modules) that depend on installed optional packages. For example, `sage.libs.coxeter3.coxeter`.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -103,12 +103,24 @@ jobs:
...
         if: always() && steps.worktree.outcome == 'success' && steps.incremental.outcome != 'success'
         run: |
           set -ex
-          ./bootstrap && make doc-clean doc-uninstall sagelib-clean && git clean -fx src/sage && ./config.status && make build
+          ./bootstrap && make doc-clean doc-uninstall sagelib-clean && git clean -fx src/sage && ./config.status
+        working-directory: ./worktree-image
+        env:
+          MAKE: make -j2
+          SAGE_NUM_THREADS: 2
+
```
Comment (kwankyu) [inline .github/workflows/build.yml:116]: Shouldn't this step be placed before the step "Build and test modularized distributions"?

File: .github/workflows/build.yml (hunk 1 above)
Comment (kwankyu) [reply .github/workflows/build.yml:124]: Thanks for the explanation.

File: .github/workflows/build.yml (hunk 2 above)
Comment (kwankyu) [reply .github/workflows/build.yml:116]: I worried that modularized distributions made in the step "Build and test modularized distributions" could be wiped out by the "Clean" step. But they would not. The "Clean" step only cleans up the sage library. Right?

Comment (kwankyu) [reply .github/workflows/build.yml:116]: This is irrelevant. Never mind.

**Review by kwankyu — APPROVED** [review]
OK. Now it looks good to me.

**Comment by mkoeppe** [comment]
Thanks!

**Pull request merged**